### PR TITLE
feat(kore): implement basic ts interpreter for Kore 2022

### PR DESCRIPTION
**Pull request by paradite**
This is a very basic interpreter for Kore 2022 in TypeScript, leveraging on the existing `kaggle-environments` cli.

I supposed this is not very useful as a standalone cli, but it can be a good starting point / reference to writing training code.

Current features:
- `run` mode (for basic training)
- support exactly 2 agents
- ability to specify number of episodes
- built-in concurrency (default 5)
- friendly statistics at end of the run
- access to `Board` object after each turn

Features not implemented yet:
- support for other number of agents
- `evaluate` mode
- `step` mode (for better training controls)
- proper cli args handling (might need to introduce 3rd party libraries)

Sample output:
```
$ npm run interpreter

> interpreter
> node --require ts-node/register interpreter.ts 2 ./main.py simple

Running 2 episodes with agents: ./main.py simple
1/2
[epi 1][step 000] current player kore:500 action 0: undefined reward: 500 status: ACTIVE
[epi 1][step 100] current player kore:34 action 0: SPAWN_1 reward: 34 status: ACTIVE
[epi 1][step 200] current player kore:125 action 0: LAUNCH_2_W reward: 126 status: ACTIVE
[epi 1][step 300] current player kore:253 action 0: undefined reward: 254 status: ACTIVE
[epi 1][step 317] current player kore:253 action 0: undefined reward: -84 status: DONE
2/2
[epi 0][step 000] current player kore:500 action 0: undefined reward: 500 status: ACTIVE
[epi 0][step 100] current player kore:55 action 0: SPAWN_1 reward: 55 status: ACTIVE
[epi 0][step 200] current player kore:236 action 0: LAUNCH_2_W reward: 236 status: ACTIVE
[epi 0][step 300] current player kore:617 action 0: SPAWN_1 reward: 617 status: ACTIVE
[epi 0][step 347] current player kore:369 action 0: SPAWN_2 reward: -54 status: DONE
game 00 rewards: -54, 37409
game 01 rewards: -84, 6743
agent main wins: 00/02
done

```

@bovard let me know what you think. I am happy to continue working on it if there are further changes needed.

### CODE REVIEW

File: kaggle_environments/envs/kore_fleets/starter_bots/ts/interpreter.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+const userLogfilename = myArgs[5];
+if (!userLogfilename) {
+  console.log('Please provide a filename for the log. Example:');
+  console.log(example);
+  process.exit(1);
+}
+
+const userResultfilename = myArgs[6];
+if (!userResultfilename) {
```
Comment (bovard) [inline kaggle_environments/envs/kore_fleets/starter_bots/ts/interpreter.ts:42]: could this be optional and default to out.log?

File: kaggle_environments/envs/kore_fleets/starter_bots/ts/interpreter.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+const agentNames = myArgs.slice(1, 5);
+// TODO: support other number of agents
+if (agentNames.length !== 4) {
+  console.log('Please provide exactly 4 agents. Example:');
+  console.log(example);
+  process.exit(1);
+}
+
+const userLogfilename = myArgs[5];
+if (!userLogfilename) {
```
Comment (bovard) [inline kaggle_environments/envs/kore_fleets/starter_bots/ts/interpreter.ts:35]: Could this be optional and default to out.log?

File: kaggle_environments/envs/kore_fleets/starter_bots/ts/interpreter.ts (hunk 1 above)
Comment (bovard) [inline kaggle_environments/envs/kore_fleets/starter_bots/ts/interpreter.ts:42]: optional and default to replay.json?

File: kaggle_environments/envs/kore_fleets/starter_bots/ts/package.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
     "test": "mocha --require ts-node/register test/**/*.ts",
     "compile": "tsc",
     "package": "tsc && tar -czvf submission.tar.gz main.py dist/*",
+    "interpreter": "node --require ts-node/register interpreter.ts 2 ./main.py simple simple attacker out.log replay.json",
```
Comment (bovard) [inline kaggle_environments/envs/kore_fleets/starter_bots/ts/package.json:14]: add a description of what this does in the README

**Comment by paradite** [comment]
@bovard updated per requests. let me know if you found any other issues.

**Comment by paradite** [comment]
@bovard should be ready for review now.

Added some sample code on how to get turn information. Sample output:

```
$ npm run interpreter

> interpreter
> node --require ts-node/register interpreter.ts 2 ./main.py simple simple attacker

Running 2 episodes with agents: ./main.py simple simple attacker
1/2
[epi 0][step 000] kore 0:0 action 0: undefined reward: 500 status: ACTIVE
[epi 0][step 100] kore 0:0 action 0: SPAWN_1 reward: 44 status: ACTIVE
[epi 0][step 200] kore 0:0 action 0: undefined reward: 51 status: ACTIVE
2/2
[epi 1][step 000] kore 0:0 action 0: undefined reward: 500 status: ACTIVE
[epi 1][step 100] kore 0:0 action 0: SPAWN_1 reward: 11 status: ACTIVE
[epi 1][step 200] kore 0:0 action 0: LAUNCH_2_W reward: 193 status: ACTIVE
[epi 1][step 300] kore 0:0 action 0: undefined reward: -149 status: DONE
game 00 rewards: -162, -246, -120, 12174
game 01 rewards: -149, -224, 7787, 14836
agent main wins: 00/02, top-twos: 00/02
done
```

**Comment by paradite** [comment]
@bovard i made some (final) update to use the `Board` class and also changed the player number to 2. please help to review again. thanks!

**Review by bovard — APPROVED** [review]
Thanks!

**Pull request merged**